cirq/experiments/cross_entropy_benchmarking.py

```diff
@@ -0,0 +1,374 @@
+    possible bit-strings in the Hilbert space.
+
+    3) Compute an experimental XEB function for each circuit_mn:
+
+    $f_{mn}^{meas} = \langle D * P^{th, mn}_q - 1 \rangle$
+
+    where D is the number of states in the Hilbert space, $P^{th, mn}_q$ is the
+    theoretical probability of a bit-string q at the end of circuit_mn, and
+    $\langle \rangle$ corresponds to the ensemble average over all measured
+    bit-strings.
+
+    Then, take the average of $f_{mn}^{meas}$ over all circuit_mn with fixed
+    n to obtain:
+
+    $f_{n} ^ {meas} = (\sum_m f_{mn}^{meas}) / M$
+
+    4) Compute a theoretical XEB function for each circuit_mn:
+
+    $f_{mn}^{th} = D \sum_q (P^{th, mn}_q) ** 2 - 1$
+
+    where the summation goes over all possible bit-strings q in the Hilbert
+    space.
+
+    Similarly, we then average $f_m^{th}$ over all circuit_mn with fixed n to
+    obtain:
+
+    $f_{n} ^ {th} = (\sum_m f_{mn}^{th}) / M$
+
+    5) Calculate the XEB fidelity $\alpha_n$ at fixed n:
+
+    $\alpha_n = f_{n} ^ {meas} / f_{n} ^ {th}$
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubits: The qubits included in the XEB experiment.
+        benchmark_ops: A sequence of ops.Moment containing gate operations
+            between specific qubits which are to be benchmarked for fidelity.
+            If more than one ops.Moment is specified, the random circuits
+            will rotate between the ops.Moment's. As an example,
+            if benchmark_ops = [Moment([ops.CZ(q0, q1), ops.CZ(q2, q3)]),
+            Moment([ops.CZ(q1, q2)]) where q0, q1, q2 and q3 are instances of
+            Qid (such as GridQubits), each random circuit will apply CZ gate
+            between q0 and q1 plus CZ between q2 and q3 for the first cycle,
+            CZ gate between q1 and q2 for the second cycle, CZ between q0 and
+            q1 and CZ between q2 and q3 for the third cycle and so on. If
+            None, the circuits will consist only of single-qubit gates.
+        num_circuits: The total number of random circuits to be used.
+        repetitions: The number of measurements for each circuit to estimate
+            the bit-string probabilities.
+        cycles: The different numbers of circuit layers in the XEB study.
+            Could be a single or a collection of values.
+        scrambling_gates_per_cycle: If None (by default), the single-qubit
+            gates are chosen from X/2 ($\pi/2$ rotation around the X axis),
+            Y/2 ($\pi/2$ rotation around the Y axis) and (X + Y)/2 ($\pi/2$
+            rotation around an axis $\pi/4$ away from the X on the equator of
+            the Bloch sphere). Otherwise the single-qubit gates for each layer
+            are chosen from a list of possible choices (each choice is a list
+            of one or more single-qubit gates).
+
+    Returns:
+        A CrossEntropyResult object that stores and plots the result.
+    """
+    simulator = sim.Simulator()
```

Comment:
probably a simulator should be passed as a parameter and it defaults to this one.  That way you can override with a custom simulator.

Reply:
Thanks for the suggestion. It's done.